src/raster/create_options.rs

```diff
@@ -0,0 +1,7 @@
+use crate::cpl::CslStringList;
+
+/// Key/value pairs of options for passing driver-specific creation flags to
+/// [`Driver::create_with_band_type_with_options`](crate::Driver::create_with_band_type_with_options`).
+///
+/// See `papszOptions` in [GDAL's `Create(...)` API documentation](https://gdal.org/api/gdaldriver_cpp.html#_CPPv4N10GDALDriver6CreateEPKciii12GDALDataType12CSLConstList).
+pub type RasterCreationOptions = CslStringList;
```

Comment:
Not 100% sure if

a) A type alias is a good idea
b) It should be in a separate file

Rationale:

a) If we ever needed to convert it into a newtype (to add creation-specific methods like `validate`, it's less breaking.
b) Putting a single type in `mod.rs` feels unbalanced.

Reply:
> a) If we ever needed to convert it into a newtype (to add creation-specific methods like validate, it's less breaking. Also, it's easy to get create options and other options confused, so this makes it a bit more clear.

We should probably re-export it from the parent module, so not breaking.

> b) Putting a single type in mod.rs feels unbalanced.

There might be more :slightly_smiling_face:.

Reply:
> We should probably re-export it from the parent module, so not breaking.

It is: 

https://github.com/georust/gdal/pull/519/files#diff-eb172c9758a6ca0ee77f152d6a8c864de3a68ea11439bb26fa33ea60d3490155R78